src/main/java/com/exadel/discountwebapp/history/service/HistoryService.java

```diff
@@ -0,0 +1,43 @@
+package com.exadel.discountwebapp.history.service;
+
+import com.exadel.discountwebapp.history.enums.OperationsEnum;
+import com.exadel.discountwebapp.history.mapper.HistoryMapper;
+import com.exadel.discountwebapp.history.vo.HistoryRequestVO;
+import com.exadel.discountwebapp.history.vo.HistoryResponseVO;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import org.javers.core.Javers;
+import org.javers.repository.jql.QueryBuilder;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class HistoryService {
+    private final Javers javers;
+    private final HistoryMapper historyMapper;
+
+    @SneakyThrows
+    public List<HistoryResponseVO> getHistory(HistoryRequestVO request) {
+
+        OperationsEnum[] operation = OperationsEnum.values();
+
+        List<HistoryResponseVO> rez = null;
+
+        for (OperationsEnum operations : operation) {
```

Comment:
It seems like the logic is incorrect here. Actually, we should take the operation type from the HistoryRequestVO. And use it in "switch-case". And there is no need in this for-each iteration "for (OperationsEnum operations : operation) {"

Reply:
changed